woocommerce/qit-cli
# Allow to wait for test completion when running tests

This PR allows running a test in sync mode by passing the `--wait` flag.

This is important if we want to allow users to delegate their testing execution to us in their own PRs, for instance.

### How to test this:
- Make sure you are connected to a Manager in the branch `23-01/results-iframe`
- `./qit run:activation {foo} --wait`
- See that the console waits for the completion of the test
- In a new terminal session, run `./qit list-tests` and see the last test ID
- Go to the Manager, enter the container with `make enter`
- Run `wp post meta update {TEST_ID} cd_status failed`
- Go back to the terminal that was "waiting", see that it shows the completed test result.
- Run `echo $?` and assert that the exit status is `1`
- Repeat the process, now setting the result of the test to `success`
- Run `echo $?` and assert that the exit status is `0`
- If you wish, you can repeat for `warning`, which exits with a `2`, and any other status is a `3`.
- Bonus: Do the same, but now with `--json` as well.

### Unhappy paths:
- Run with a timeout (minimum is 10), eg: `./qit run:activation {foo} --wait --timeout=10`, see that the execution ends after the timeout is reached with a meaningful message informing that the test is still running, but we've reached the timeout for waiting for it.
- Run without a timeout, but instead of updating the test result on the Manager, abort it with Ctrl+C, assert that you receive a meaningful message that the test is still running, but we've aborted waiting for it.
